src/helpers/parser.ts

```diff
@@ -49,38 +40,69 @@ export const gitLinkedIssueParser = async ({ owner, repo, pull_number }: GitPars
   }
 };
 
-export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser) => {
+export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser): Promise<HTMLElement[]> => {
   const logger = getLogger();
   try {
     const { data } = await axios.get(`https://github.com/${owner}/${repo}/issues/${issue_number}`);
-    const context = getBotContext();
-    const payload = context.payload as Payload;
     const dom = parse(data);
     const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
     const linkedPRs = devForm.querySelectorAll(".my-1");
-    if (linkedPRs.length === 0) return null;
+    if (linkedPRs.length === 0) return [];
+    return linkedPRs;
+  } catch (error) {
+    logger.error(`${JSON.stringify(error)}`);
+    return [];
+  }
+};
+
+// Get the linked PRs from the issue, if latest is true, return the lastest merged PR
+export const getLinkedPrs = async ({ owner, repo, issue_number, latest }: GitParser): Promise<PRsForClose[] | ListPullsByNumberResponse["data"] | null> => {
+  const logger = getLogger();
+  const context = getBotContext();
+  const payload = context.payload as Payload;
+  try {
+    const linkedPRs = await gitLinkedPrParser({ owner, repo, issue_number });
+    const prs: PRsForClose[] = [];
     let linkedPullRequest = null;
+
     for (const linkedPr of linkedPRs) {
-      const prHref = linkedPr.querySelector("a")?.attrs?.href || "";
-      const parts = prHref.split("/");
-      // extract the organization name and repo name from the link:(e.g. "https://github.com/wannacfuture/ubiquibot/pull/5";)
+      const prUrl = linkedPr.querySelector("a")?.attrs?.href;
+
+      if (!prUrl) continue;
+
+      const parts = prUrl.split("/");
+      // extract the organization name and repo name from the link:(e.g. "
       const organization = parts[parts.length - 4];
       const repository = parts[parts.length - 3];
 
       if (`${organization}/${repository}` !== payload.repository.full_name) continue;
-      const prNumber = parts[parts.length - 1];
-      if (Number.isNaN(Number(prNumber))) return null;
-      const pr = await getPullByNumber(context, Number(prNumber));
-      if (!pr || !pr.merged) continue;
 
-      if (!linkedPullRequest) linkedPullRequest = pr;
-      else if (linkedPullRequest.merged_at && pr.merged_at && new Date(linkedPullRequest.merged_at) < new Date(pr.merged_at)) {
-        linkedPullRequest = pr;
+      const prNumber = Number(parts[parts.length - 1]);
+
+      if (latest) {
```

Comment:
like I said this functionality should be in calling functions, the only purpose of this function should be to get linked PRs

Reply:
You can view the file in full to fully get it, there's the function you want to be generic  and there's another to get useful information from the parsed data

Reply:
the function should parse html and return array of PRs, other things should be done by the calling function - no need for latest functionality in this function